### DISCUSSION TITLE
Can't take a photo of my educational document on laptop or phone camera

### DISCUSSION THREAD

**Discussion by mikeffendii** [community/community#41401] [GitHub Education] [answered]
Greetings!

I was apply to the GitHub student developer pack but got stuck in **uploading educational proof**. My phone and laptop camera don't work and there's no other option for uploading the educational documents. 

I want the GitHub support to email me **an upload link** for my document verification soon so I can start utilizing the student developer pack.

Thank you

**Comment by samus-aran** [comment]
Hi @mikeffendii 

First off I want to thank you for sharing your experiences and that we absolutely acknowledge your frustration. We are trying our best to continually improve our system to be as easy as possible for legitimate students everywhere to verify their academic status, while also ensuring that bad actors impersonating students don't jeopardise our ability to provide academic benefits of value.

As you can imagine there are many types of academic verification and institutes around the world and our system is constantly evolving to accommodate this without setting the bar too high. However sometimes we may require a little more evidence from the student applying and we thank you for being patient with it.

It would be preferable that a student is on campus applying and thus able to upload a document but unfortunately with remote learning that isn't always possible and so we request students submit a live photo of their evidence rather than an upload. I understand that your phone may not be the best of quality but as long as the text is legible the system should be able to process it. Heres are a few tips on using a photo to capture.

1. Ensure your image (photo or upload) is legible. If you are unable to read its text on the image itself then the verification process may not be able to either. If you are taking a photo of a screen try to have it as large a resolution as possible - take a photo of the evidence on a monitor (a laptop perhaps - or a TV) rather than a phone.
2. If you can, print off your evidence and take a photo of this copy.
3. Keep the text legible. It's preferred to have the full shot of the evidence but try and capture the important sections of it to confirm your student status. If the image is too large and makes the text tiny then zoom in as appropriately.

All applications must go through GitHub Global Campus [Student Application form](https://education.github.com/discount_requests/student_application). We are unable to provide an upload link for individual applicants.

**Reply by meierms1** [reply]
This verification system you have is completely trash. Are you saying that a technology company like GitHub can not manage to flip between front and back cameras? This was posted 6 months ago and the back camera is still not working. Is it supposed to be a challenge where you have to print the documents and then take a picture with the front camera while still keeping the dates legible?

**Reply by fahmizainal17** [reply]
Could you guys help me get two answer accepted?

**Reply by ashutoshrai387** [reply]
All your points are valid , but not even in the slightest manner it clarifies that why only front camera is allowed for live capture and not back camera. Your response is totally pointless.

**Reply by SamarthRajj** [reply]
well it still does not work

**Reply by MBanatovski** [reply]
still doesn't

**Reply by SamarthRajj** [reply]
i think they don't want it to work. it's like with the govt website make a feature unusable so it is still there but you can't use it

**Reply by RRDMELO** [reply]
by God because I can't use the rear camera. It's impossible for me to angle a good quality photo with a front camera if I can't see the photo.
sincerely. I think this is a flaw in the system that needs to be resolved.

**Reply by mgupta70** [reply]
After trying to switch cameras probably 30-40 times,  I got the rear camera once. But after 1 photo, it changed back to the front camera. Plus, I need to justify why I am not on campus, even though I am. Please look into it.

**Reply by jestes15** [reply]
I tried swapping cameras on my iPhone and GitHub would not let me use anything except my selfie camera. I tried using my desktop camera with DATES and it still rejected this shit. Sincerely, fix your shit.

**Reply by JulioRios00** [reply]
Same here. Why can't we just upload the document?

**Reply by jayshozie** [reply]
It automatically picks the worst camera possible, too. I cannot possibly get a legible and readable picture of the documentation. Just allow us to upload the document itself. It was done that way before. If you don't want to, then just don't mislead us.

**Reply by MaxwellJpg** [reply]
Not sure if this is still an issue for any of you, but I found a solution - see my [other comment](https://github.com/orgs/community/discussions/41401?sort=new#discussioncomment-15017818)

**Reply by PoSihDun** [reply]
> Not sure if this is still an issue for any of you, but I found a solution - see my [other comment](https://github.com/orgs/community/discussions/41401?sort=new#discussioncomment-15017818)

I love you sooooo much!

**Reply by HillGnur** [reply]
@MaxwellJpg solution worked for me! Thx

**Reply by ghabriel-fiorotti** [reply]
@MaxwellJpg, You're not Ronaldo, but you're a phenomenon.

**Reply by pat104471** [reply]
Why would they use a camera to upload documentation and not just file uploads?

**Reply by yasin-miyade** [reply]
Why it's always use front camera on Laptop/Desktop/Mobile, Front camera can't capture clear photo and it's unable to use back camera for clear photo.

**Comment by Kel-VinTech** [comment]
You just have to get a clear view of the image you ant to upload. You can upload directly with your phone or snap and send to your laptop then upload it.
Either way should work

**Reply by Ayush-Kumar-web** [reply]
no it does not work @Kel-VinTech  I rejected my request saying that uploaded images can be manipulated easily upload from device camera 
but it does not allow taking photos from back camera

**Reply by fahmizainal17** [reply]
Could you guys help me get two answer accepted?

**Comment by mikeffendii** [comment]
Hey Kevin.
There's no button for upload. There's a single button for taking picture
and as I click that, my laptop camera opens, so how can I take a picture of
my documents through the laptop camera? And by doing it through phone, my
phone's front camera opens so how can I can take a clear picture of my
document through the front camera? I need to get a link for uploading my
document in pdf format or a rare camera link through which I can take a
clear picture.
Thank you

**Reply by samus-aran** [reply]
Thanks @mikeffendii for your feedback.
 
All applications must go through GitHub Global Campus [Student Application form](https://education.github.com/discount_requests/student_application). We are unable to provide an upload link for individual applicants.

When you take a photo on your phone do you get an option to select back & front? Are you able to take a screenshot from your phone when this happens (or doesn't)?

**Reply by A373k** [reply]
back and front both  gives front camera option only please make an upload file option

**Reply by RRDMELO** [reply]
when you select rear camera,
nothing simply happens. continues on the front. I already tested it on a Samsung smartphone. iphone 15, xiaomi, motorola.
Does not enable rear camera.

**Reply by ignaciostellino** [reply]
Same here , can you please fix this?

**Comment by mikeffendii** [comment]
Nope, I'm not getting option for front or back camera. The front camera
opens by default and I can't take a picture of my document with front
camera. Kindly help me with this documents uploading process.

**Reply by samus-aran** [reply]
Hi @mikeffendii 

Can you take a screenshot of the application form (via the phone) when you get to the 'take a photo' section.

**Comment by mikeffendii** [comment]
The screenshot is attached to this email, kindly have a look at it and
please help me with this application process so I can get the student pack
soon.

**Reply by samus-aran** [reply]
Hi @mikeffendii 

There doesn't seem to be an image attached to your reply. You're able to upload the screenshot directly to your reply by clicking on the "attach files by..."

![Can-t-take-a-photo-of-my-educational-document-on-laptop-or-phone-camera-·-Discussion-41401-·-community](https://user-images.githubusercontent.com/4041275/207735522-f6dc167b-c3b0-47c4-b24c-10ae981af16d.png)

**Reply by iamkelv** [reply]
> Hi @mikeffendii
> 
> There doesn't seem to be an image attached to your reply. You're able to upload the screenshot directly to your reply by clicking on the "attach files by..."
> 
> ![Can-t-take-a-photo-of-my-educational-document-on-laptop-or-phone-camera-·-Discussion-41401-·-community](https://user-images.githubusercontent.com/4041275/207735522-f6dc167b-c3b0-47c4-b24c-10ae981af16d.png)

can you help me out? I have been having the same issue here is my document. Thank you.

Moderator Edit: Removing PII

**Reply by samus-aran** [reply]
Hi @iamkelv 

I've removed your private information from [this post](https://github.com/orgs/community/discussions/41401#discussioncomment-4757641) as posting identifiable information is not allowed, per our [Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md).

All applications must go through [GitHub Global Campus Student Application form](https://education.github.com/discount_requests/student_application). We are unable to provide manual reviews or upload links for individual applicants.

**Reply by fahmizainal17** [reply]
Could you guys help me get two answer accepted?

**Comment by mikeffendii** [comment]
I can't see this option on my phone neither on my laptop. Which browser
should I use for it?

**Reply by samus-aran** [reply]
Hi @mikeffendii 

Here are a few ways to add images to a reply.


1. View this discussions page via your mobile browser. There should be an option via your browser settings to select "desktop site". This will change the screen layout to be more like the desktop. When making a reply you there will be the "Attach files by dragging & dropping, selecting or pasting them" bar along the bottom similar to the screenshot I posted above. Documentation can be found here - https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files

2. View this discussions page via your mobile browser. When you begin a reply there is an editor bar along the top (in both mobile and smaller desktop views). The icon that looks like an image (a mountain + sun) when selected will ask you to take a photo (which isn't relevant to this case) or select an existing image on your phone (or desktop).
![Screenshot_github_addimagr](https://user-images.githubusercontent.com/4041275/207938339-f46a1cd8-13c4-44d8-99ce-4de664f80448.png)


Hope this helps and allows us to see what you see when applying to GitHub Global Campus. 👍

**Reply by hw88hw88** [reply]
My uploaded valid documents have been rejected for more than 10 times.
The camera does not work.

**Reply by fahmizainal17** [reply]
Could you guys help me get two answer accepted?

**Comment by mikeffendii** [comment]
I tried Google Chrome, Microsoft Edge, and Brave browser for it but no
browser worked well and I didn't see the option to upload the documents.

**Reply by Kel-VinTech** [reply]
Can you try with with another phone or laptop

**Reply by iamkelv** [reply]
It's not working. Seeing the same issue

**Reply by mikeffendii** [reply]
I tried it on different browsers on my phone and laptop, but nothing worked out. 
The option to upload a picture is only available here in the community discussion. So are we supposed to upload our educational documents here in order to get the GitHub student developer pack?

**Reply by SamarthRajj** [reply]
One year has passed it still doesn't work

**Comment by pewf** [comment]
This is still happening. Do not ignore this.

**Reply by hw88hw88** [reply]
> This is still happening. Do not ignore this.

I can't find the solution. Please GitHub support e-mail me a link to upload my document or review my rejected submission.
There is no point to reject my submissions. Any one of my submissions alone is sufficient to verify my student status.

**Reply by pewf** [reply]
Yea. I've made 14 submissions so far. I'm honestly stuck but I'm going to keep trying

**Comment by Kira-Pgr** [comment]
I can upload the picture, But I stuck at processing application. It stays on the same page after processing, and still ask me to "Upload proof of your academic status" I've tried serveral devices and browsers

**Comment by real-jiakai** [comment]
This issue is still existed. My classmate wants to upload the proof of his academic status. But the safari in iPad can't use the back camera, only the front camera works.

**Reply by fahmizainal17** [reply]
Could you guys help me get two answer accepted?

**Comment by xsheen7** [comment]
Can you teach me how to use th front camera to take a photo of proof of academic status? Why not back camera?

**Comment by dayidp** [comment]
same issue

**Comment by LRSnowX** [comment]
same issue

**Comment by fahmizainal17** [comment]
Could you guys help me get two answer accepted?

**Comment by ayush222006** [comment]
THIS IS FOR ANYONE who can't seem to use their back camera in taking photo.


First have the image of your document in your laptop/computer
Then start obs studio 
Then add those pictures in obs studio and click start virtual camera on the right hand side of menu.

Then go to GitHub student verification page and then click take photo obs studio virtual camera will be selected by default.

Then take photo and upload it.
It will be done 👍.

**Reply by malevitzch** [reply]
Doesn't work for me, sayt that "Please use your device camera to submit your academic affiliation documents."

**Comment by fahmizainal17** [comment]
Great thank you

**Comment by lequan81** [comment]
This is for anyone who has this issue on computer.

Just simply take a photo of your student ID on your phone.
Then send that image through your computer (via Drive, OneDrive, etc).
After that, drag and drop that image into the box "Take picture" and if you see the progress bar showing uploading your image, you are good to go.

**Comment by ee0806898683** [comment]
HBI

**Comment by ee0806898683** [comment]
### 

> _****_

**Comment by TheTeaCup** [comment]
i am having a similar issue but for me i have the option to select my back camera but when i do nothing appears. 
![IMG_4425](https://github.com/community/community/assets/35646791/21aa095e-a7fd-4ea2-b29d-3fffeb6dee3f)

**Comment by MertcanAlgan** [comment]
Guys I think i found the solution. But you need a mac and an Iphone. 
1. Connect your mac to a monitor
2. close the screen of your mac
3. connect your iphone to your mac via a cable
4. then when you take a picture of your student id it will use the camera on your phone

**Comment by SamarthRajj** [comment]
it is like they don't want to make it work

**Comment by zubairjammu786** [comment]
Same issue. Very frustrated by this.   There should be a upload button files to upload. They just need verification not the way how we upload

**Comment by soufianemou** [comment]
the same problem i tried it with different browsers but in vaint the back camera didn't work

**Reply by zubairjammu786** [reply]
Its fraustrating

**Reply by andersonszdc** [reply]
this is unbelievable

**Reply by zubairjammu786** [reply]
Its the most useless feature on Github. We are students but we can't get the access to student developers pack

**Reply by zubairjammu786** [reply]
Hey recently i was going with these issues. You can read my comments above. But i've found a solution.
You can use a **VIRTUAL CAMERA.** (even if you don't have a physical camera ). In my case i used e2esoft vcam but you can use anyone like OBS Studio.
So open your virtual camera software and select your image for proof verification. Then in your browser open apply for student developer pack application. Enter all details and click on take photo button for documens. It'll automatically capture your selected image in Virtual camera. So in this way you can get your application approved, It works. Let's have a try

**Comment by zubairjammu786** [comment]
Hey recently i was going with these issues. You can read my comments above. But i've found a solution.
You can use a **VIRTUAL CAMERA.** (even if you don't have a physical camera ). In my case i used e2esoft vcam but you can use anyone like OBS Studio.
So open your virtual camera software and select your image for proof verification. Then in your browser open apply for student developer pack application. Enter all details and click on take photo button for documens. It'll automatically capture your selected image in Virtual camera. So in this way you can get your application approved, It works. Let's have a try

**Reply by timothy-mckenna** [reply]
this seems like a viable workaround to a profoundly stupid process. Apparently, GitHub has never heard of email.
Go to your school so you connect to GitHub from the school's IP adpdress, or become a hacker who can spoof IP packets. Loading the GitHub page through the school's VPN should satisfy them but doesn't.
OBS Studio:  set Sources to Display Capture, select and frame display, Start Virtual Camera
Reload GitHub's idiot "Upload proof of your academic status" page.
take a photo with the "OBS Virtual Camera"
Good luck not wasting your time.

**Comment by Wolfyxon** [comment]
![image](https://github.com/community/community/assets/58263600/d81c8d4e-41f1-4779-bc44-5c94fbb4e3e4)

For me after allowing access to the camera and clicking **Take a picture**, there's just this loading icon. I've tried both on my phone and PC. I've used the file upload method before to apply (seems like it's no longer available)

Is there anything I can do? I need to re-apply.

**Reply by zubairjammu786** [reply]
try from another browser and make sure your browser has access to your camera. you can also try virtual camera. i've written a detailed comment above. you can check that

**Reply by Wolfyxon** [reply]
I tried on multiple browsers and devices.

**Reply by zubairjammu786** [reply]
try virtual camera. maybe it'll work for you

**Reply by Wolfyxon** [reply]
I tried, the thing is pretty much nothing happens after clicking **Take picture**, there's just this infinitely spinning loading icon.

**Comment by zubairjammu786** [comment]
Then create a new github account.

**Comment by Wolfyxon** [comment]
I fixed it on mobile. The camera permission was denied in the app permissions so Chrome didn't ask me to allow GitHub to use my camera.

However on PC the camera works properly and GitHub is allowed to use it. The indicator lamp turns on after clicking **Take picture**, yet nothing happens. This is the case with both physical and virtual cameras.

I will apply using my phone, but this should be fixed on PC too.

**Reply by zubairjammu786** [reply]
Great, If you can do it with mobile phone, then go for it. All the best

**Comment by Wolfyxon** [comment]
Another issue. School IDs in my country don't have the expiration date on the front and I'm only able to attach a single photo. The expiration date is required. What do I do?
 When there was still an option to upload a file I just combined 2 photos of both sides.

**Reply by zubairjammu786** [reply]
i don't think expiration date is required. Moreover you can also submit your transcript ( result card ). If your school id is valid it'll be accepted

**Reply by Wolfyxon** [reply]
![image](https://github.com/community/community/assets/58263600/3af2fe3c-1315-4f5e-97d4-95d366f66880)

It is required, I've seen other people's IDs get rejected because of the same issue as me.
But I've figured out the issue on my PC, it was a problem with OBS and GitHub refused to start using the cameras if one of them wasn't working. I'll put both pictures in OBS and hopefully it will be accepted.

**Reply by zubairjammu786** [reply]
Great

**Comment by hadil-jilani** [comment]
same issue here

**Comment by ruhi47** [comment]
Hey Guys,
Try to read through what the rejection reasons are.

- If you see something like this : You must configure your browser and operating system to allow location access. You may not use a VPN. If you believe that our information about your school is incomplete or incorrect, then please select ‘My selected school has incorrect or incomplete information e.g. domains or campus location’ option when submitting a GitHub Education support ticket. - Try to allow the browser to access location and if it asks "why aren't you on campus?" - Suppose, in my case it was distance learning on slack, I selected distance learning option and clicked a picture of my slack channel on phone with PC.

- If you see something like this : Your image contains keywords indicating that it may be an acceptance letter. Only proof of current academic enrollment is acceptable. - you might have tried uploading acceptance letter, please try to upload last unofficial transcript you have either from website or received on email.

- If you see something like this : Please ensure that your academic affiliation document contains your first name exactly as it appears in your [GitHub billing information](https://github.com/settings/billing/payment_information). Variations of your name and nicknames are unacceptable. - Make sure when you go on Github> Settings> Payment> Billing Information , the name registered on School (thus printed on transcript) MUST match the name in the Billing Information. 

- Apart from this you must have:
1. Verify your academic email address on your GitHub account, if your school provides one.
2. Secure your GitHub account with TFA. (I used Microsoft Authenticator)
3. GitHub Profile must have a profile picture of yours, your name, your pronouns, and more.
4. Generate a README for your public GitHub Profile and publish it for the world to see.

Hope this helps!

P.S. While taking the photo make sure it is readable when clicking it a little blur won't effect the results.
Comment down if you need more help.

Thank you!

**Reply by ken-code-wq** [reply]
Please help me with the fourth step

**Reply by ruhi47** [reply]
Hey @ken-code-wq , what do you need help with in specific?

**Reply by ken-code-wq** [reply]
Yh @ruhi47 thanks for getting back, I was getting a prompt that asked why I wasn't on campus and I did not have any certified document to prove my vacation and reopening date. But when I went back to campus it worked.

Thanks Github

**Reply by adithiyaaboi** [reply]
@ruhi47 If you see something like this : Please ensure that your academic affiliation document contains your first name exactly as it appears in your [GitHub billing information](https://github.com/settings/billing/payment_information). Variations of your name and nicknames are unacceptable.  I  did put the exact name  still it says the same its the 13th time i have applied ;-;

**Comment by RRDMELO** [comment]
I went to my campus and asked for all the documents, student card. registration certificate. school history, etc., etc. I tried 17 times, and each time I was denied, always the same answer that was not possible to verify. Look, I expected more from github. All my classmates in the class managed to register, I was awarded. Thanks GITHUB, thank you very much. for nothing.

**Comment by mirzaabdullaht** [comment]
<img width="215" alt="image" src="https://github.com/user-attachments/assets/a96dcf8d-88ad-4588-b586-1cb95ab8f776">


Hey everyone,
I wanted to share my experience with getting the GitHub Student Developer Pack. I ran into the same problem many of you are facing - there was no option to upload documents, and when I tried using the live camera, my application kept getting rejected.

Here's what I did:

I tried applying from different devices
I emailed GitHub's customer support
I even used a MacBook because its camera is way better than other laptops
My student ID was a bit tricky because it had two pages - one with my name and the other with the expiry date. After a bunch of rejections and following their suggestions, I gave it another shot with the MacBook. It took 5 rejections over a few days, but on my 6th try, it finally worked!

Now I'm just waiting for the benefits to kick in, which should take about 3 days.

Some tips from my experience:

Don't give up. Keep trying until you get approved.
Make sure to apply before you graduate. I only had 2 months left on my student ID.
Try to get the clearest picture possible of your ID.
I know how frustrating it is when you're trying to upload a clear document but can't, and you keep getting rejected. But hang in there!

My GitHub education request finally got approved.

Good luck to all of you trying to get your GitHub Education Pack :)

**Comment by RRDMELO** [comment]
Hello, friends, I got my subscription after 20 frustrating attempts. but I
had to contact my educational department and ask for an email from the
institution. and I sent documentation along with it and it worked..
Now I'm just happy. :)

Em sex., 8 de nov. de 2024 às 09:41, Abdulah ***@***.***>
escreveu:

**Comment by Hackerking237** [comment]
Muje hacking sekhni ha 😭😭😭😭

**Comment by Riffles** [comment]
How is this still an issue? It's almost 2025. You guys are a freaking coding company.. Wtf .... Use your own copilot and fix this.... Jesus Christ...

**Reply by ken-code-wq** [reply]
Thats true tho

**Reply by DtotheAtotheVtotheE** [reply]
This is so damn frustrating....I'm about to say screw it and just eat ramen and go pay for another AI.  Like what in the hell?

**Comment by anujpatel03** [comment]
I am also facing same issue. There is not any upload button and when I am trying this in my phone then it is taking picture from the front camera only. This is making difficulty in applying.

**Comment by laszlogellen** [comment]
Hey everyone! Use Opera to fix the issue. Something is wrong with Chrome. (I have android)

**Reply by megamissil** [reply]
Opera on iOS didn't work

**Comment by megamissil** [comment]
This is STILL an issue. Are you guys okay?

**Reply by enzovill94** [reply]
same this is still an issue for me. I cannot seem to use the back camera of my phone, and the front camera is so bad...

**Reply by oyvinds78** [reply]
Me as well. This is just stupid.

**Comment by doflare123** [comment]
I encountered a problem that you can't attach a file/photo on the site, you can only take one. I don't have a camera on my computer, and the front camera opens on my phone, which can't take a clear photo, what should I do?

**Comment by Cadeyrnn** [comment]
Quite unbelievable that a site meant to be used by the worlds biggest programming community could be so behind on a feature like this. It is like it is intentionally nerfed. I have tried from all my devices on multiple browsers including OBS studio set at highest resolution. As soon as I hit capture its a pixelated piece of rubbish that of course gets denied. What is stopping Github from using authorisation by signing in through the education portal like basically any other software on the planet?

**Comment by onisankochira** [comment]
I applied at the beginning of this school year, got denied. Tried again, denied again. Now I can't even try anymore because even though I uploaded my statement of enrollment pdf screenshot, there's this red "!" preventing me from submitting the application. This system is messed and stupid. I don't think they want to give me approval.

**Comment by setday** [comment]
Hm, I tried the obs method, but it didn't work because the camera was just empty. But I found an alternative solution that should work on any laptop.

To do this, you need to change one line in the element code.

<img width="960" height="787" alt="Screenshot 2025-07-28 234202" src="https://github.com/user-attachments/assets/9e77ad09-175a-4baf-ab8d-126ac1213d74" />
Open the item code (F12) -> click on the item selection icon (on the top left of the open panel in the form of an arrow pointing at the screen) -> select the "Take a photo" item -> find the line shown in the screenshot and replace the value "..."allowFileUpload":false..." with "..."allowFileUpload":true..."

At first, nothing will change, but after refreshing the page and filling out the form, you will see a photo upload button.

**Reply by abu3beid** [reply]
didnt work :(

**Comment by mekchou** [comment]
Let's be honest, this is a scam, and GitHub just wants everyone to pay the bill. Period.

**Reply by MaxwellJpg** [reply]
hey, not sure if this is still an issue for you, but I found a solution - see my [other comment](https://github.com/orgs/community/discussions/41401?sort=new#discussioncomment-15017818)

**Reply by bigb45** [reply]
@MaxwellJpg man what a great workaround, thanks!

**Reply by jamesondollaman-create** [reply]
Just go away , your destroying me & laugh all you want , I do have 3 adult kids & 1 is a cop so ... you keep this up allgood , 
I'm sure we have people in our worlds you wanted to be stalked , harassed,  laugh at me all you want ... it's not just me this effect & surely would you want your Aka , family ( no Intentions ) just see it from another side of coin .
A learning lesson...check yourself,  I DIDNT DO THIS TO MYSELF ...It all started because someone had a cry about the world k owing the real them ..so give it up ..LAST TIME & ILL SAY PLEASE

**Comment by MaxwellJpg** [comment]
Work around for the file upload button not being present.

EDIT: for TLDR

Run the following in the console (CTRL + SHIFT + J)
```javascript
document.querySelector("#dev_pack_form_camera_required").value = 'false';
document.querySelector('#js-developer-pack-application-submit-button').click()
```
Which will set the value and resubmit the form for it to take affect - button should then be present

Full details on what this does below from my original post that walks through it manually with explanations.

----

Inspect the page and within the form there is a number of `d-none` elements.

Locate the one regarding 'camera required' (~4th element) and set the value from true to false
<img width="987" height="218" alt="image" src="https://github.com/user-attachments/assets/bf4dc2e8-c5c3-40d9-a58f-ceeaaebab7e1" />

Once set to true we need to reload the form but with this value...

I found that as part of the validation, the data the user adds for these fields hits the server and when there are validation issues the data is sent back as it came. Meaning that if we fail the validation the modal and form will reload but importantly with this camera required field set to what we want.

So, the next step is to simple submit the form blank.

Once submitted you will get the file is required message, but now with the upload button present.

From this:
<img width="1111" height="1381" alt="image" src="https://github.com/user-attachments/assets/153dd292-208d-4e8d-a635-5f92f882e052" />

to this:
<img width="1113" height="1386" alt="image" src="https://github.com/user-attachments/assets/9ccb4f8f-6f56-4c7b-b98b-b68218948470" />

Also bear in mind that only image file types are supported.

successful with upload
<img width="1114" height="392" alt="image" src="https://github.com/user-attachments/assets/0966d8db-4476-4ac3-b5be-b57dce701b97" />


_update showing this working_
<img width="1464" height="219" alt="image" src="https://github.com/user-attachments/assets/f8976c04-b4c0-4dbd-82ed-3ffe83b21c65" />


I hope this helps!
Lets get this comment higher to help others get around this intentional restriction.

**Reply by patricks-devhub** [reply]
don't work

**Reply by bsofiasandoval** [reply]
worked for me on google chrome for the file uploading fix as of today :) Thank you so much for the fix!
<img width="704" height="86" alt="Screenshot 2025-12-05 at 9 25 55 PM" src="https://github.com/user-attachments/assets/fe90f078-00cc-461c-b112-6b199136857e" />

**Reply by Shelvak** [reply]
Thanks man, best hack ever!! 🔥🔥🔥🔥🔥🔥🔥🔥🔥

(Tested in firefox)

Run in console:
- `document.querySelector("#dev_pack_form_camera_required").value = 'false'`
- Submit the form (to get the error)
- Upload the image



<img width="1664" height="336" alt="imagen" src="https://github.com/user-attachments/assets/8540c14b-2129-4eea-9c3f-328e0b5ca275" />

**Reply by reckten** [reply]
Worked for me, thanks!

**Reply by DaMarcusKeetonWorks** [reply]
Worked like a charm, thank you.

STEPS: (in Chrome at least)
1. Right-click and  inspect 

2. Use the select an element to select down by the Start camera button to get you close, then within the form, there are a number of d-none elements.  - - -Locate the one regarding 'camera required' (~4th element) and set the value from true to false

3. hit submit application, and the box will reload with the option to upload a document ( make sure it's an image Doc - Jpeg, jpg, png)

**Reply by PoSihDun** [reply]
Bro your heaven sent <3

GitHub - hire this person!

**Reply by MaxwellJpg** [reply]
Glad this is working!
I've edited the post to have a console method that should be quicker, thanks to @Shelvak for the suggestion.

**Reply by marcintosh87** [reply]
You are amazing, thank you.

**Reply by N3M3S1Spy** [reply]
Bro, your a Gigachat

**Reply by yonetimkurulu** [reply]
Thanks, thanks, thanks king of the github.

**Reply by petraamaral** [reply]
YOU ARE INSANE!

**Reply by BlueOutside** [reply]
I think github patched this exploit server-side

<img width="1113" height="305" alt="image" src="https://github.com/user-attachments/assets/1de1e7d3-0c4a-436b-bf07-403465af3712" />
_(No I do NOT have a school mail, it's only for teachers)_
Atp I'll just give up

**Reply by chaincert** [reply]
This worked for me today, I took a screenshot of the enrollment proof and saved it as Jpg. Then uploaded it, and it was approved nearly instantly, it's an automated system scanning the photos of the documents, Just follow the above instructions and upload a photo of your document as jpg.
![Screenshot 2026-04-11 105615](https://github.com/user-attachments/assets/2a309f79-9885-4d24-ad57-bcbdc732ad40)

**Reply by Supakorn21** [reply]
Hi, You are life saver!! Thanks for this.
<img width="929" height="333" alt="Screenshot 2026-04-17 at 11 20 01" src="https://github.com/user-attachments/assets/ecd21d29-6b86-45c0-9de7-4c4b863cd0a7" />

**Reply by jonathanrietz** [reply]
Still works in May 2026! This is so silly, lol.

**Reply by kahei-li** [reply]
Has this been patched now? I am getting this rejection message now :(
<img width="1000" height="362" alt="image" src="https://github.com/user-attachments/assets/8193c7f9-9912-4dcc-9e66-a49ef6bd2ffd" />

**Reply by jamesalbeza** [reply]
We are too late man.

**Reply by Razielsach** [reply]
Is it already too late for this?

**Reply by vijaylabx** [reply]
<img width="1080" height="1603" alt="Screenshot_20260917_150031_Chrome" src="https://github.com/user-attachments/assets/8db1ac90-4f7a-42ab-bfbb-629f84f4fdf5" />
Can anyone help me 🙏 for approve my profile

**Reply by zouzzz-zzzuoz** [reply]
Look here : https://github.com/orgs/community/discussions/41401#discussioncomment-18052920

**Reply by vijaylabx** [reply]
> Look here : https://github.com/orgs/community/discussions/41401#discussioncomment-18052920

<img width="1156" height="497" alt="github_Student_benefits" src="https://github.com/user-attachments/assets/a88925a4-1fd9-4fa3-bffd-aca99302af61" />
finally i goted this for your help Thank You

**Reply by MaxwellJpg** [reply]
The link from @zouzzz-zzzuoz is a literal copy of my OP, that you are all replying to. Can you not just read that?

**Comment by mrdulal** [comment]
Give this guy medal @MaxwellJpg. Thank you bro. You save the day. :-)

**Reply by codaire** [reply]
It seems Github has blocked this approach, making it more clear that the camera limitation is a feature, not a glitch.

**Reply by Mudityadev** [reply]
hey, facing same issue but it's worked with android phone, chrome browser. hope it helps

**Comment by TLucking** [comment]
WHY IS IT NOT POSSIBLE TO SIMPLY UPLOAD A PDF?? This is so stupid to print out the pdf to take a picture of it

**Reply by mayu-z** [reply]
it does that sometimes, make sure your proof has the date of you joining the school/college and some basic info like college's name and logo, and address as well.

Since you're using a pdf take a screenshot of the pdf like the content, make sure it has your photo and everything else mentioned above and if it has the time of your graduation then it'll be better, sometimes your college isn't mentioned in github make sure you choose "add college".

If something is troubling your feel free to reply to me.

If this resolves it an upvote would be appreciated.

**Reply by TLucking** [reply]
This doesn't answer the question of what I should do with my pdf, since there is no upload pdf button?! My problem is that the webpage requires me to take a picture with my camera of my documents instead of simply being able to upload them. Theses are just stupid unnecessary steps. 

I have all needed documents but no printer. Why does github require to print out the pdfs to take a picture of them instead of just allowing to upload them via an upload link????

**Reply by enablaaaa-zouzzz** [reply]
Same pb, I can't upload...

**Comment by mayu-z** [comment]
it does that sometimes, make sure your proof has the date of you joining the school/college and some basic info like college's name and logo, and address as well.

Since you're using a pdf take a screenshot of the pdf like the content, make sure it has your photo and everything else mentioned above and if it has the time of your graduation then it'll be better, sometimes your college isn't mentioned in github make sure you choose "add college".

If something is troubling your feel free to reply to me.

If this resolves it an upvote would be appreciated.

**Reply by paksubham357** [reply]
i tried taking photo with laptop camera as well as mobile camera of my college ID. but it's still getting rejected. i hav filled billing details and turned ON 2FA. even updated my profile details as well.  whta else is required to get verified ? help me out here please

**Comment by bochkarev-artem** [comment]
The quality of a photo will be degraded (on purpose?), so its impossible to get validated. Seems like a scam option from github that just doesnt want to give any benefits to the students :)

**Comment by enablaaaa-zouzzz** [comment]
https://bidouillouzzz.blogspot.com/2026/04/github-copilot-education-comment-faire.html

**Reply by vintotal** [reply]
Je confirme que ça fonctionne sur Microsoft Edge en septembre 2026. :)
<img width="422" height="419" alt="image" src="https://github.com/user-attachments/assets/aa35591d-420d-45e5-befe-0b427d13e056" />

**Comment by enablaaaa-zouzzz** [comment]
Eh eh cool. Merci du feedback.
Mais bon va n'empêche pas que Github-Copilot est malhonnête sur ce coup.

**Comment by ps2builds** [comment]
This is a useful GitHub Community discussion, especially for users who may be dealing with the same issue. Clear details about the affected feature, expected behavior, and any workaround or solution found would make the thread even more helpful for others troubleshooting it.

**Comment by samus-aran** [comment, marked as answer]
We're closing this discussion because this original issue has evolved over the years and this doesn't translate to current camera issues. Thank you for helping us maintain a productive and tidy community for all our members.

If you wish to ask for help in verification feel free to open up a new discussion.
